examples/jsm/helpers/LightProbeHelperGPU.js

```diff
@@ -0,0 +1,63 @@
+import {
+	Mesh,
+	NodeMaterial,
+	SphereGeometry
+} from 'three';
+import { float, Fn, getShIrradianceAt, normalWorld, uniformArray, uniform, vec4 } from 'three/tsl';
+
+class LightProbeHelper extends Mesh {
+
+	constructor( lightProbe, size = 1 ) {
+
+		const sh = uniformArray( lightProbe.sh.coefficients );
+		const intensity = uniform( lightProbe.intensity );
+
+		const RECIPROCAL_PI = float( 1 / Math.PI );
+
+		const fragmentNode = Fn( () => {
+
+			const irradiance = getShIrradianceAt( normalWorld, sh );
+
+			const outgoingLight = RECIPROCAL_PI.mul( irradiance ).mul( intensity );
+
+			return vec4( outgoingLight, 1.0 );
+
+		} )();
+
+		const material = new NodeMaterial();
+		material.fragmentNode = fragmentNode;
+
+		const geometry = new SphereGeometry( 1, 32, 16 );
+
+		super( geometry, material );
+
+		this.lightProbe = lightProbe;
+		this.size = size;
+		this.type = 'LightProbeHelper';
+
+		this._intensity = intensity;
```

Comment:
Is this property doing anything?

Reply:
The uniform is updated in `onBeforeRender()`.

Reply:
I do see that, but why is the same "hack" not needed for the probe coefficients, but is needed for the intensity?

Reply:
The line is indeed redundant. I'll remove it.

Reply:
>The line is indeed redundant. I'll remove it.

This PR was working. Now it is not.

If you figure it out, please do your best to answer my question above so we will both understand. 🙏

Reply:
What do you mean with not working?

Reply:
The `_ intensity` property is removed in #29303.

Reply:
Ah wait, I see the issue now. #29303 should fix the issue.

Reply:
It doesn't respond to changes in the light probe intensity -- but it did before you deleted the line.

Reply:
I think there is a difference because intensity is a primitive value and _sh an array of objects. It probably would not work if you actually exchange the vector objects and not just update their values.

Reply:
I think we can rewrite the intensity uniform like so:
```js
const intensity = reference( 'intensity', 'float', lightProbe );
```
Then a manual uniform update isn't necessary. However, `reference()` would not work for uniform arrays ,afaik.

Reply:
You added `_sh` in #29303, and that was not necessary if you are updating just the contents of the array.

With all due respect, I think you merged PRs too fast, without allowing time for the reviewer to respond.

Can you please continue to investigate this so we can all understand?

Reply:
> You added _sh in https://github.com/mrdoob/three.js/pull/29303, and that was not necessary if you are updating just the contents of the array.

I suspect you have tested this by just updating the _existing_ vector objects, right? However, the array is overwritten with new objects, the changes from https://github.com/mrdoob/three.js/pull/29303 are required. 

I just wanted to be on the safe side, that's why I have honored the coefficients in https://github.com/mrdoob/three.js/pull/29303. If you think this is not required since the coefficient vectors are not overwritten with new instances, https://github.com/mrdoob/three.js/pull/29303 can be partly reverted. 

> With all due respect, I think you merged PRs too fast, without allowing time for the reviewer to respond.

I understand but it is also imported to move forward with the migration. We can fix such minor issue in `dev` as well.

Reply:
>you have tested this by just updating the existing vector objects, right?

Right. Just the contents.

>If you think this is not required since the coefficient vectors are usually not overwritten with new instances, https://github.com/mrdoob/three.js/pull/29303 can be partly reverted.

I think it's not required. Granted, it is a judgment call.